### ISSUE TITLE
Add support for ENS off-chain resolver

### ISSUE THREAD

**Issue by bonustrack** [snapshot-labs/snapshot-hub#474]
We need to add support for ENS offchain resolver, this has been used by Coinbase and recently Lens Protocol to give their user an ENS domain without paying fee for each of them, see more info https://docs.ens.domains/dapp-developer-guide/ens-l2-offchain and https://github.com/ensdomains/offchain-resolver

**Comment by findtoni** [comment]
I'll work on this

**Comment by bonustrack** [comment]
@findtoni No, it wont work this way, what we want is to make it possible to use an ENS domain coming from offchain resolver for example `fabien.lens.xyz` https://app.ens.domains/name/fabien.lens.xyz to create a space at https://snapshot.org/#/fabien.lens.xyz The space wont issue subdomain, this is done automatically on others protocols like Lens.

**Comment by findtoni** [comment]
@bonustrack Oh I get it now, thanks

**Comment by bonustrack** [comment]
It may not be possible for the user to update the ENS `snapshot` text record directly on his subdomain, in this case we may want to allow creating a space from the owner of the subdomain (the user).

**Comment by eswarasai** [comment]
Would like to pick this up, thanks!

**Comment by samuveth** [comment]
@findtoni can you share progress or open a draft PR. Otherwise I would suggest to give it to @eswarasai

**Comment by findtoni** [comment]
Okay, @eswarasai can have it.

**Comment by zzuziak** [comment]
Hey @eswarasai 👋  What's the status of this issue, are you still up for solving it?
